[IMP] delivery_purchase_label: Label Stock Move Field to Source PO Line

raumschmiede-joshuaL opened the pull request:
Hi.  

It is complicated to get from the label picking stock move to the corresponding PO line in code,With a BOM in the PO and its components in the moves it is even more complicated to have a loop to search for the right line.

With this new field it is easier. And the label picking behaves more like a normal picking

@TDu , @mt-software-de

mt-software-de commented on delivery_purchase_label/models/stock_move.py:
```diff
@@ -0,0 +1,21 @@
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
```
```suggestion
class StockMove(models.Model):
```

mt-software-de commented on delivery_purchase_label/models/stock_move.py:
```diff
@@ -0,0 +1,21 @@
+    )
+
+    # Prevent the lines with distinct label PO lines from being merged to have an
+    # exact connection between them
```
```suggestion
    # Prevent merging delivery label moves
    # fom different purchase order lines
```

mt-software-de reviewed (changes requested):
Thx. For your contribution. 
The problem here is not a third party system. It is the fact that it should not be allowed to merge delivery label moves from different purchase order lines. 
This ensures that the Delivery label Transfer hast the equal moves as the normal purchase order transfer. 
Accordingly to that could please Update the title of the PR and your commit Message.

mt-software-de commented on delivery_purchase_label/__manifest__.py:
```diff
@@ -1,9 +1,11 @@
 # Copyright 2024 Camptocamp SA
+# Copyright 2025 Raumschmiede GmbH
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
```
```suggestion
```

mt-software-de commented on delivery_purchase_label/__manifest__.py:
```diff
@@ -1,9 +1,11 @@
     "name": "Delivery Purchase Label",
     "summary": "Allows printing carrier delivery labels for a dropshipping vendor.",
-    "version": "14.0.1.0.0",
+    "version": "14.0.1.1.0",
```
This should be done by a psc when merging. 
```suggestion
```

raumschmiede-joshuaL committed:
[IMP] delivery_purchase_label: Add link between delivery label move and purchase order line
And prevent merging delivery label moves with moves from different purchase order lines

mt-software-de commented on delivery_purchase_label/__manifest__.py:
```diff
@@ -1,4 +1,5 @@
 # Copyright 2024 Camptocamp SA
+# Copyright 2025 Raumschmiede GmbH
```
```suggestion
```

mt-software-de commented on delivery_purchase_label/models/stock_move.py:
```diff
@@ -0,0 +1,21 @@
+    )
+
+    # Prevent merging delivery label moves
+    # fom different purchase order lines
```
```suggestion
    # from different purchase order lines
```

mt-software-de reviewed (changes requested):
Some minor remarks.

mt-software-de reviewed (approved):
LGTM. ~Could you please update your commit message.~
@jbaudoux Could you please review.

mt-software-de commented:
cc @i-vyshnevska

jbaudoux commented:
/ocabot merge minor
